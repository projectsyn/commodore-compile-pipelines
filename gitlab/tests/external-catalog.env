CLUSTERS="c-cluster-id-1234 c-cluster-id-5678 c-cluster-id-1111"
CLUSTER_CATALOG_URLS="c-cluster-id-1234=ssh://git@git.vshn.net/cluster-catalogs/c-cluster-id-1234.git c-cluster-id-5678=ssh://git@git.example.com/cluster-catalogs/c-cluster-id-5678.git c-cluster-id-1111=https://git.vshn.net/cluster-catalogs/c-cluster-id-1111.git"
