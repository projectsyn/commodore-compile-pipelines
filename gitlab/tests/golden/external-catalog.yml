{
   "c-cluster-id-1111_compile": {
      "artifacts": {
         "expire_in": "1 week",
         "paths": [
            "diff.txt"
         ]
      },
      "before_script": [
         "install --directory --mode=0700 ~/.ssh",
         "echo \"$SSH_KNOWN_HOSTS\" >> ~/.ssh/known_hosts",
         "echo \"$SSH_CONFIG\" >> ~/.ssh/config"
      ],
      "image": {
         "name": "docker.io/projectsyn/commodore:v1.23.0"
      },
      "rules": [
         {
            "if": "$CI_COMMIT_REF_NAME != $CI_DEFAULT_BRANCH"
         }
      ],
      "script": [
         "git config --global url.\"https://gitlab-ci-token:${CI_JOB_TOKEN}@git.vshn.net:80\".insteadOf ssh://git@${CI_SERVER_SHELL_SSH_HOST}",
         "git config --add --global url.\"https://${ACCESS_USER_c_cluster_id_1111:-token}:${ACCESS_TOKEN_c_cluster_id_1111}@git.vshn.net/cluster-catalogs/c-cluster-id-1111.git\".insteadOf https://git.vshn.net/cluster-catalogs/c-cluster-id-1111.git",
         "git config --add --global url.\"https://${ACCESS_USER_c_cluster_id_1111:-token}:${ACCESS_TOKEN_c_cluster_id_1111}@git.vshn.net/cluster-catalogs/c-cluster-id-1111.git\".insteadOf ssh://git@git.vshn.net/cluster-catalogs/c-cluster-id-1111.git",
         "/usr/local/bin/entrypoint.sh commodore catalog compile --tenant-repo-revision-override $CI_COMMIT_SHA c-cluster-id-1111",
         "(cd catalog/ && git --no-pager diff --staged --output ../diff.txt)"
      ],
      "stage": "build",
      "variables": {
         "KUBERNETES_CPU_LIMIT": "2",
         "KUBERNETES_CPU_REQUEST": "800m",
         "KUBERNETES_MEMORY_LIMIT": "3Gi"
      }
   },
   "c-cluster-id-1111_deploy": {
      "before_script": [
         "install --directory --mode=0700 ~/.ssh",
         "echo \"$SSH_KNOWN_HOSTS\" >> ~/.ssh/known_hosts",
         "echo \"$SSH_CONFIG\" >> ~/.ssh/config"
      ],
      "image": {
         "name": "docker.io/projectsyn/commodore:v1.23.0"
      },
      "rules": [
         {
            "if": "$CI_COMMIT_REF_NAME == $CI_DEFAULT_BRANCH"
         }
      ],
      "script": [
         "git config --global url.\"https://gitlab-ci-token:${CI_JOB_TOKEN}@git.vshn.net:80\".insteadOf ssh://git@${CI_SERVER_SHELL_SSH_HOST}",
         "git config --add --global url.\"https://${ACCESS_USER_c_cluster_id_1111:-token}:${ACCESS_TOKEN_c_cluster_id_1111}@git.vshn.net/cluster-catalogs/c-cluster-id-1111.git\".insteadOf https://git.vshn.net/cluster-catalogs/c-cluster-id-1111.git",
         "git config --add --global url.\"https://${ACCESS_USER_c_cluster_id_1111:-token}:${ACCESS_TOKEN_c_cluster_id_1111}@git.vshn.net/cluster-catalogs/c-cluster-id-1111.git\".insteadOf ssh://git@git.vshn.net/cluster-catalogs/c-cluster-id-1111.git",
         "/usr/local/bin/entrypoint.sh commodore catalog compile --push c-cluster-id-1111"
      ],
      "stage": "deploy",
      "variables": {
         "KUBERNETES_CPU_LIMIT": "2",
         "KUBERNETES_CPU_REQUEST": "800m",
         "KUBERNETES_MEMORY_LIMIT": "3Gi"
      }
   },
   "c-cluster-id-1234_compile": {
      "artifacts": {
         "expire_in": "1 week",
         "paths": [
            "diff.txt"
         ]
      },
      "before_script": [
         "install --directory --mode=0700 ~/.ssh",
         "echo \"$SSH_KNOWN_HOSTS\" >> ~/.ssh/known_hosts",
         "echo \"$SSH_CONFIG\" >> ~/.ssh/config"
      ],
      "image": {
         "name": "docker.io/projectsyn/commodore:v1.23.0"
      },
      "rules": [
         {
            "if": "$CI_COMMIT_REF_NAME != $CI_DEFAULT_BRANCH"
         }
      ],
      "script": [
         "git config --global url.\"https://gitlab-ci-token:${CI_JOB_TOKEN}@git.vshn.net:80\".insteadOf ssh://git@${CI_SERVER_SHELL_SSH_HOST}",
         "git config --global url.\"https://gitlab-ci-token:${ACCESS_TOKEN_c_cluster_id_1234}@git.vshn.net:80/cluster-catalogs/c-cluster-id-1234.git\".insteadOf ssh://git@${CI_SERVER_SHELL_SSH_HOST}/cluster-catalogs/c-cluster-id-1234.git",
         "/usr/local/bin/entrypoint.sh commodore catalog compile --tenant-repo-revision-override $CI_COMMIT_SHA c-cluster-id-1234",
         "(cd catalog/ && git --no-pager diff --staged --output ../diff.txt)"
      ],
      "stage": "build",
      "variables": {
         "KUBERNETES_CPU_LIMIT": "2",
         "KUBERNETES_CPU_REQUEST": "800m",
         "KUBERNETES_MEMORY_LIMIT": "3Gi"
      }
   },
   "c-cluster-id-1234_deploy": {
      "before_script": [
         "install --directory --mode=0700 ~/.ssh",
         "echo \"$SSH_KNOWN_HOSTS\" >> ~/.ssh/known_hosts",
         "echo \"$SSH_CONFIG\" >> ~/.ssh/config"
      ],
      "image": {
         "name": "docker.io/projectsyn/commodore:v1.23.0"
      },
      "rules": [
         {
            "if": "$CI_COMMIT_REF_NAME == $CI_DEFAULT_BRANCH"
         }
      ],
      "script": [
         "git config --global url.\"https://gitlab-ci-token:${CI_JOB_TOKEN}@git.vshn.net:80\".insteadOf ssh://git@${CI_SERVER_SHELL_SSH_HOST}",
         "git config --global url.\"https://gitlab-ci-token:${ACCESS_TOKEN_c_cluster_id_1234}@git.vshn.net:80/cluster-catalogs/c-cluster-id-1234.git\".insteadOf ssh://git@${CI_SERVER_SHELL_SSH_HOST}/cluster-catalogs/c-cluster-id-1234.git",
         "/usr/local/bin/entrypoint.sh commodore catalog compile --push c-cluster-id-1234"
      ],
      "stage": "deploy",
      "variables": {
         "KUBERNETES_CPU_LIMIT": "2",
         "KUBERNETES_CPU_REQUEST": "800m",
         "KUBERNETES_MEMORY_LIMIT": "3Gi"
      }
   },
   "c-cluster-id-5678_compile": {
      "artifacts": {
         "expire_in": "1 week",
         "paths": [
            "diff.txt"
         ]
      },
      "before_script": [
         "install --directory --mode=0700 ~/.ssh",
         "echo \"$SSH_KNOWN_HOSTS\" >> ~/.ssh/known_hosts",
         "echo \"$SSH_CONFIG\" >> ~/.ssh/config"
      ],
      "image": {
         "name": "docker.io/projectsyn/commodore:v1.23.0"
      },
      "rules": [
         {
            "if": "$CI_COMMIT_REF_NAME != $CI_DEFAULT_BRANCH"
         }
      ],
      "script": [
         "git config --global url.\"https://gitlab-ci-token:${CI_JOB_TOKEN}@git.vshn.net:80\".insteadOf ssh://git@${CI_SERVER_SHELL_SSH_HOST}",
         "/usr/local/bin/entrypoint.sh commodore catalog compile --tenant-repo-revision-override $CI_COMMIT_SHA c-cluster-id-5678",
         "(cd catalog/ && git --no-pager diff --staged --output ../diff.txt)"
      ],
      "stage": "build",
      "variables": {
         "KUBERNETES_CPU_LIMIT": "2",
         "KUBERNETES_CPU_REQUEST": "800m",
         "KUBERNETES_MEMORY_LIMIT": "3Gi"
      }
   },
   "c-cluster-id-5678_deploy": {
      "before_script": [
         "install --directory --mode=0700 ~/.ssh",
         "echo \"$SSH_KNOWN_HOSTS\" >> ~/.ssh/known_hosts",
         "echo \"$SSH_CONFIG\" >> ~/.ssh/config"
      ],
      "image": {
         "name": "docker.io/projectsyn/commodore:v1.23.0"
      },
      "rules": [
         {
            "if": "$CI_COMMIT_REF_NAME == $CI_DEFAULT_BRANCH"
         }
      ],
      "script": [
         "git config --global url.\"https://gitlab-ci-token:${CI_JOB_TOKEN}@git.vshn.net:80\".insteadOf ssh://git@${CI_SERVER_SHELL_SSH_HOST}",
         "/usr/local/bin/entrypoint.sh commodore catalog compile --push c-cluster-id-5678"
      ],
      "stage": "deploy",
      "variables": {
         "KUBERNETES_CPU_LIMIT": "2",
         "KUBERNETES_CPU_REQUEST": "800m",
         "KUBERNETES_MEMORY_LIMIT": "3Gi"
      }
   }
}
